packages/blade/src/components/BottomSheet/BottomSheet.native.tsx

```diff
@@ -55,9 +55,9 @@ const _BottomSheet = ({
   const bottomSheetAndDropdownGlue = useBottomSheetAndDropdownGlue();
   const defaultInitialFocusRef = React.useRef<any>(null);
   const sheetRef = React.useRef<GorhomBottomSheet>(null);
-  const header = React.useRef<React.ReactNode>();
-  const footer = React.useRef<React.ReactNode>();
-  const body = React.useRef<React.ReactNode>();
+  const [header, setHeader] = React.useState<React.ReactNode>();
+  const [footer, setFooter] = React.useState<React.ReactNode>();
+  const [body, setBody] = React.useState<React.ReactNode>();
```

Comment:
Won't this cause lots of re-rendering? Why did we change it to state from ref?

Reply:
> Why did we change it to state from ref?

This bug was happening because we saved the children of the BottomSheetBody, BottomSheetFooter & BottomSheetHeader in a ref and then rendered those refs which meant that any updates to the children components inside BottomSheetBody will only reflect if the parent component rerenders.

> Won't this cause lots of re-rendering? 

It *should* rerender when parent children changes.

Reply:
Why do we need to put it in a state and then pass it down to the bottomsheet? Why can't we just directly pass it down from the props?

Reply:
Because GorhomBottomsheet requires it to be on each prop eg `footerComponent`, `handleComponent` and the body needs to be on the `children` prop. 

And our JSX is structured as follows: 

```jsx
                <BottomSheet
                  isOpen={isOpen}
                  onDismiss={() => {
                    setIsOpen(false);
                  }}
                >
                  <BottomSheetHeader title="Terms & Conditions" subtitle="Read carefully before accepting." />
                  <BottomSheetBody>
                    <Text>
                      Lorem Ipsum i</Text>
                  </BottomSheetBody>
                  <BottomSheetFooter>
                    ....
                  </BottomSheetFooter>
                </BottomSheet>
```

So we need to extract our BottomSheetBody, BottomSheetFooter, BottomSheetHeader and put them on the correct gorhom props.